### ISSUE TITLE
Handle more complex SASS/SCSS syntax

### ISSUE THREAD

**Issue by babakness** [mrmckeb/typescript-plugin-css-modules#17]
**Is your feature request related to a problem? Please describe.**

Yes, the css incompatible parts of `sass` don't seem to be supported.

It would be great if we could write with programmatic short hand like this in SASS

```scss
@for $section from 1 to 10 {
  .section-#{$section} {
    // something systematic
  }
} 
```

and get typings for the compiled CSS result

```css
.section-1 { ... }
.section-2 { ... }
.section-3 { ... }
```

**Describe the solution you'd like**
SASS/SCSS files are compiled for full SASS pre-processor language support 

**Describe alternatives you've considered**
n/a

**Comment by babakness** [comment]
My naive approach was to try to compile SASS files to CSS before classes are extracted. However, the compilation is asynchronous (callback). It does not appear that this function

https://github.com/mrmckeb/typescript-plugin-css-modules/blob/769366a64321fa7961f0e1e9f6f523c46237a264/src/index.ts#L19

Can be `async`  or return a `Promise`. Not too familiar with writing language servers for VSCode.

BTW, I tried using `renderSync` on both `node-sass` and `sass` and they both seem to crash the language server in VS Code.

I also tried a couple different `postcss-sass` plugins, didn't seem to work either.

**Comment by babakness** [comment]
For now, I've made a package to compile sass into type definitions for use in projects

<img src="https://raw.githubusercontent.com/babakness/sass-module-types/master/.github/images/animation.gif" alt="sass-module-types animation" />

https://github.com/babakness/sass-module-types

**Comment by mrmckeb** [comment]
Thanks @babakness, sorry for the slow response - I've had a busy week...

This has been a hard exercise for me too, as there obviously needs to be a balance between feature support and speed of compilation... the more features supported, the slower updates will be - which impacts TypeScript's performance too.

I'm not entirely sure what the best path forward is here. I'm currently working on something for Create React App, which is eating my open source time, but could look at this in 2 weeks or so.

That module looks cool, I'll definitely reference it when adding improvements here.

**Comment by hellocontrol-bng** [comment]
+1 on this, especially because using those complex features actually prevents the rest of the file from working as expected. I have a file that breaks because it uses a mixin, and the result is that NONE of the classes in that file are reflected, even if they are not using the mixin at all.

**Comment by DavidBabel** [comment]
Your both libs are awesome guys. You can make a pretty neat thing if you guys work together ;)

**Comment by mrmckeb** [comment]
Closed by #20. Will be released later today :)
